src/developer-quickstart.md

```diff
@@ -0,0 +1,441 @@
+
+    #[storage(read)]
+    fn count() -> u64; 
+}
+```
+
+### Going line by line
+
+`#[storage(read, write)]` is an annotation which denotes that this function has permission to read and write a value in storage.
+
+`fn increment();` - We're introducing the functionality to increment and denoting it shouldn't return any value.
+
+`#[storage(read)]` is an annotation which denotes that this function has permission to read values in storage.
+
+`fn counter() -> u64;` - We're introducing the functionality to increment the counter and denoting the function's return value.
+
+### Implement ABI
+
+Below your ABI definition, you will write the implementation of the functions defined in your ABI.
+
+```sway
+impl Counter for Contract {
+    #[storage(read)]
+    fn count() -> u64 {
+      return storage.counter;
+    }
+    #[storage(read, write)]
+    fn increment(){
+        storage.counter = storage.counter + 1;
+    }
+}
+```
+
+> **Note**
+> `return storage.counter;` is equivalent to `storage.counter`.
+
+### What we just did
+
+Read and return the counter property value from the contract storage.
+
+```sway
+fn count() -> u64 {
+    return storage.counter;
+}
+```
+
+The function body accesses the value counter in storage, and increments the value by one. Then, we return the newly updated value of counter.
+
+```sway
+fn increment() {
+    storage.counter = storage.counter + 1;
+}
+```
+
+Here's what your code should look like so far:
+
+```sway
+contract;
+
+storage {
+    counter: u64 = 0,
+}
+
+abi Counter {
+    #[storage(read, write)]
+    fn increment();
+
+    #[storage(read)]
+    fn count() -> u64;
+}
+
+impl Counter for Contract {
+    #[storage(read)]
+    fn count() -> u64 {
+      return storage.counter;
+    }
+    #[storage(read, write)]
+    fn increment(){
+        storage.counter = storage.counter + 1;
+    }
+}
+```
+
+### Build the Contract
+
+Build `counter_dapp` by running the following command in your terminal from inside the `counter_dapp` directory:
+
+```sh
+forc build
+```
+
+You should see something like this output:
+
+```console
+Compiled library "core".
+  Compiled library "std".
+  Compiled contract "counter_dapp".
+  Bytecode size is 240 bytes.
+```
+
+## Testing your Contract
+
+In the directory `tests`, navigate to `harness.rs.` Here you'll see there is some boilerplate code to help you start interacting with and testing your contract.
+
+At the bottom of the file, define the body of `can_get_contract_instance`. Here is what your code should look like to verify that the value of the counter did get incremented:
+
+```sway
+#[tokio::test]
+async fn can_get_contract_id() {
+    let (instance, _id) = get_contract_instance().await;
+    // Now you have an instance of your contract you can use to test each function
+    instance.increment().call().await.unwrap();
+    let result = instance.count().call().await.unwrap();
+    assert!(result.value > 0); 
+}
+```
+
+Run the following command in the terminal: `forc test`.
```

Comment:
As a part [of our next Sway Tooling milestone](https://github.com/orgs/FuelLabs/projects/14/views/3), the `forc test` command will be reworked to behave as a command for in-language *unit testing* (rather than Rust-only integration testing). You can see [this issue](https://github.com/FuelLabs/sway/issues/1833) for some context. [This WIP PR](https://github.com/FuelLabs/sway/pull/2264) which I hope to land next week begins the process by removing the integration testing support in favour of doing Rust integration testing via `cargo` (e.g. using `cargo test` directly instead).

Perhaps it's worth adding a `> Note: ...` at the end of this section that says something along the lines of:

> Note: The `forc test` command is in the process of being reworked to perform in-language unit testing, at which point we will recommend using `cargo` directly for your Sway+Rust integration testing. See [this issue](https://github.com/FuelLabs/sway/issues/1833) for more details.